### PR TITLE
feat: data encoder

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Data part encoder

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/mito2/src/memtable/merge_tree/data.rs [hunk 1]
```diff
@@ -94,23 +108,302 @@ impl DataBuffer {
     }
 
     /// Freezes `DataBuffer` to bytes. Use `pk_lookup_table` to convert pk_id to encoded primary key bytes.
```
Comment (evenyag) [inline src/mito2/src/memtable/merge_tree/data.rs]:
```suggestion
    /// Freezes `DataBuffer` to bytes. Use `pk_weights` to convert pk_id to encoded primary key bytes.
```

File: src/mito2/src/memtable/merge_tree/data.rs [hunk 2]
```diff
@@ -94,23 +108,302 @@ impl DataBuffer {
...
-    pub fn freeze(self, _pk_wights: &[u16]) -> DataPart {
-        todo!()
+    pub fn freeze(self, pk_wights: &[u16]) -> Result<DataPart> {
```
Comment (evenyag) [inline src/mito2/src/memtable/merge_tree/data.rs]:
```suggestion
    pub fn freeze(self, pk_weights: &[u16]) -> Result<DataPart> {
```

Comment (evenyag) [inline src/mito2/src/memtable/merge_tree/data.rs]: `freeze()` may takes `&self`
```suggestion
    pub fn freeze(&self, pk_wights: &[u16]) -> Result<DataPart> {
```

File: src/mito2/src/memtable/merge_tree/data.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/mito2/src/memtable/merge_tree/data.rs]: fixed in https://github.com/evenyag/greptimedb/pull/8/commits/5f6cabc794a1e8b38d9f55d06523b61ca06c883e

File: src/mito2/src/memtable/merge_tree/data.rs (hunk 2 above)
Comment (v0y4g3r) [reply src/mito2/src/memtable/merge_tree/data.rs]: fixed in https://github.com/evenyag/greptimedb/pull/8/commits/5f6cabc794a1e8b38d9f55d06523b61ca06c883e

Comment (v0y4g3r) [reply src/mito2/src/memtable/merge_tree/data.rs]: fixed in https://github.com/evenyag/greptimedb/pull/8/commits/5f6cabc794a1e8b38d9f55d06523b61ca06c883e

File: src/mito2/src/memtable/merge_tree/data.rs [hunk 3]
```diff
@@ -93,24 +109,349 @@ impl DataBuffer {
         }
     }
 
-    /// Freezes `DataBuffer` to bytes. Use `pk_lookup_table` to convert pk_id to encoded primary key bytes.
-    pub fn freeze(self, _pk_wights: &[u16]) -> DataPart {
-        todo!()
+    /// Freezes `DataBuffer` to bytes. Use `pk_wights` to convert pk_id to pk sort order.
```
Comment (evenyag) [inline src/mito2/src/memtable/merge_tree/data.rs]:
```suggestion
    /// Freezes `DataBuffer` to bytes. Use `pk_weights` to convert pk_id to pk sort order.
```

**Review by evenyag — APPROVED** [review]

**Pull request merged**